src/common/libutil/base256.c

```diff
@@ -0,0 +1,149 @@
+    "👓", "🕶", "👔", "👕", "👖", "👗", "👘", "👙", "👚", "👛", "👜", "👝", "🛍", "🎒", "👞", "👟",
+    "👠", "👡", "👢", "👑", "👒", "🎩", "🎓", "📿", "💄", "💍", "💎", "🐵", "🐒", "🐶", "🐕", "🐩",
+    "🐺", "🐱", "🐈", "🦁", "🐯", "🐅", "🐆", "🐴", "🐎", "🦄", "🐮", "🐂", "🐃", "🐄", "🐷", "🐖",
+    "🐗", "🐽", "🐏", "🐑", "🐐", "🐪", "🐫", "🐘", "🐭", "🐁", "🐀", "🐹", "🐰", "🐇", "🐿", "🐻"
+}, {
+    "🐨", "🐼", "🐾", "🦃", "🐔", "🐓", "🐣", "🐤", "🐥", "🐦", "🐧", "🕊", "🐸", "🐊", "🐢", "🐍",
+    "🐲", "🐉", "🐳", "🐋", "🐬", "🐟", "🐠", "🐡", "🐙", "🐚", "🦀", "🐌", "🐛", "🐜", "🐝", "🐞",
+    "🕷", "🕸", "🦂", "💐", "🌸", "💮", "🏵", "🌹", "🌺", "🌻", "🌼", "🌷", "🌱", "🌲", "🌳", "🌴",
+    "🌵", "🌾", "🌿", "🍀", "🍁", "🍂", "🍃", "🍇", "🍈", "🍉", "🍊", "🍋", "🍌", "🍍", "🍎", "🍏",
+    "🍐", "🍑", "🍒", "🍓", "🍅", "🍆", "🌽", "🌶", "🍄", "🌰", "🍞", "🧀", "🍖", "🍗", "🍔", "🍟",
+    "🍕", "🌭", "🌮", "🌯", "🍳", "🍲", "🍿", "🍱", "🍘", "🍙", "🍚", "🍛", "🍜", "🍝", "🍠", "🍢",
+    "🍣", "🍤", "🍥", "🍡", "🍦", "🍧", "🍨", "🍩", "🍪", "🎂", "🍰", "🍫", "🍬", "🍭", "🍮", "🍯",
+    "🍼", "🍵", "🍶", "🍾", "🍷", "🍸", "🍹", "🍺", "🍻", "🍽", "🍴", "🔪", "🏺", "🌍", "🌎", "🌏",
+    "🌐", "🗺", "🗾", "🏔", "🌋", "🗻", "🏕", "🏖", "🏜", "🏝", "🏞", "🏟", "🏛", "🏗", "🏘", "🏙",
+    "🏚", "🏠", "🏡", "🏢", "🏣", "🏤", "🏥", "🏦", "🏨", "🏩", "🏪", "🏫", "🏬", "🏭", "🏯", "🏰",
+    "💒", "🗼", "🗽", "🕌", "🕍", "🕋", "🌁", "🌃", "🌄", "🌅", "🌆", "🌇", "🌉", "🌌", "🎠", "🎡",
+    "🎢", "💈", "🎪", "🎭", "🖼", "🎨", "🎰", "🚂", "🚃", "🚄", "🚅", "🚆", "🚇", "🚈", "🚉", "🚊",
+    "🚝", "🚞", "🚋", "🚌", "🚍", "🚎", "🚐", "🚑", "🚒", "🚓", "🚔", "🚕", "🚖", "🚗", "🚘", "🚙",
+    "🚚", "🚛", "🚜", "🚲", "🚏", "🛣", "🛤", "🚨", "🚥", "🚦", "🚧", "🚤", "🛳", "🛥", "🚢", "🛩",
+    "🛫", "🛬", "💺", "🚁", "🚟", "🚠", "🚡", "🚀", "🛰", "🛎", "🚪", "🛌", "🛏", "🛋", "🚽", "🚿",
+    "🛀", "🛁", "🕰", "🌞", "🌝", "🌚", "🌑", "🌒", "🌓", "🌔", "🌕", "🌖", "🌗", "🌘", "🌜", "🌛"
+}};
+
+int base256_encode (char *buf,
+                    int buflen,
+                    void *data,
+                    int datalen)
+{
+    register unsigned char table = 0;
+    int i = 0;
+    int n = 0;
+
+    if (buf == NULL || data == NULL || buflen <= 0 || datalen < 0) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    /* Copy prefix */
+    memcpy (buf, BASE64_PREFIX, strlen(BASE64_PREFIX));
+    n += strlen (BASE64_PREFIX);
+
+    while (i < datalen && n < buflen - 2) {
+        memcpy (buf + n, enc_tab[table][((unsigned char *) data)[i]], 4);
+
+        table = 1 - table;
+        i++;
+        n += 4;
+    }
+    buf[n++] = '\0';
+    return n;
+}
+
+static int base256_lookup (const char *c, char *result)
+{
+    for (int i = 0; i < 512; i++) {
+        const char *index = enc_tab[i / 256][i % 256];
+        if (memcmp (c, index, 4) == 0) {
+            *result = (char) i;
+            return 0;
+        }
+    }
+    errno = ENOENT;
+    return -1;
+}
+
+int base256_decode (void *buf, int buflen, const char *in)
+{
+    int in_size;
+    const char *cur;
+    int i = 0;
+    int n = 0;
+
+    if (buf == NULL || in == NULL || buflen <= 0) {
+        errno = EINVAL;
+        return -1;
+    }
+
```

Comment:
similar point about checking buf lens, what if length of `in` was 5, then there will be a `memcmp` of bad memory?